sympy/series/sequences.py

```diff
@@ -0,0 +1,522 @@
+from __future__ import print_function, division
+
+from sympy.core.expr import Expr
+from sympy.core.singleton import (S, Singleton)
+from sympy.core.compatibility import (range, integer_types, with_metaclass\
+                                      , is_sequence)
+from sympy.core.sympify import sympify
+from sympy.core.containers import Tuple
+from sympy.functions.elementary.integers import ceiling
+from sympy.utilities.misc import filldedent
+from sympy.sets.sets import Interval, Set
+
+
+class SeqBase(Expr):
+    """Base class for sequences"""
+
+    is_iterable = False
+
+    @property
+    def gen(self):
+        """Returns the generator for the sequence"""
+        return self._gen
+
+    @property
+    def _gen(self):
+        raise NotImplementedError("(%s)._gen" % self)
+
+    @property
+    def interval(self):
+        """The interval in which the sequence is defined"""
+        return self._interval
+
+    @property
+    def _interval(self):
+        raise NotImplementedError("(%s)._interval" % self)
+
+    @property
+    def start(self):
+        """The starting point of the sequence. This point is included"""
+        return self._start
+
+    @property
+    def _start(self):
+        raise NotImplementedError("(%s)._start" % self)
+
+    @property
+    def stop(self):
+        """The ending point of the sequence. This point is included"""
+        return self._stop
+
+    @property
+    def _stop(self):
+        raise NotImplementedError("(%s)._stop" % self)
+
+    @property
+    def step(self):
+        """Increase points by step"""
+        return self._step
+
+    @property
+    def _step(self):
+        raise NotImplementedError("(%s)._step" % self)
+
+    @property
+    def length(self):
+        """Length of the sequence"""
+        return self._length
+
+    @property
+    def _length(self):
```

Comment:
Actually, turns out `len` requires `__len__` to output something that can be an integer. So symbolic lengths won't work (this seems silly to me). So perhaps your method before was fine? @flacjacket : thoughts?